### PR TITLE
Move `network` implementation to separate package

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

We've had a circular dependency here leading to pretty confusing code. Breaking it up now ;)

## How this works

moving some stuff around + cleaned up tests

```
ok      github.com/ava-labs/avalanchego/vms/platformvm/network  0.550s  coverage: 93.2% of statements
```

## How this was tested

CI

### CODE REVIEW

File: vms/platformvm/block/builder/helpers_test.go [hunk 1]
```diff
@@ -428,3 +429,17 @@ func shutdownEnvironment(env *environment) error {
 		env.baseDB.Close(),
 	)
 }
+
+func getValidTx(txBuilder txbuilder.Builder, t *testing.T) *txs.Tx {
```
Comment (dhrubabasu) [inline vms/platformvm/block/builder/helpers_test.go:433]: This is used in `builder_test.go` and was originally in `network_test.go`. Added here as we moved `network_test`.go to its own pkg

File: vms/platformvm/network/network.go [hunk 2]
```diff
@@ -109,60 +107,80 @@ func (n *network) AppGossip(ctx context.Context, nodeID ids.NodeID, msgBytes []b
...
 	}
-
-	// add to mempool
-	if err := n.IssueTx(ctx, tx); err != nil {
-		n.ctx.Log.Debug("tx failed verification",
-			zap.Stringer("nodeID", nodeID),
-			zap.Error(err),
-		)
+	err = n.issueTx(tx)
+	n.ctx.Lock.Unlock()
```
Comment (abi87) [inline vms/platformvm/network/network.go:123]: if we don't use `defer n.ctx.Lock.Unlock()` we must unlock in every code path. Add n.ctx.Lock.Unlock() within if above, line 119

Comment (dhrubabasu) [reply vms/platformvm/network/network.go]: This didn't get reverted correctly, fixed

**Review by abi87 — APPROVED** [review]

File: vms/platformvm/network/network.go [hunk 3]
```diff
@@ -109,60 +107,81 @@ func (n *network) AppGossip(ctx context.Context, nodeID ids.NodeID, msgBytes []b
...
 	}
-
-	// add to mempool
-	if err := n.IssueTx(ctx, tx); err != nil {
-		n.ctx.Log.Debug("tx failed verification",
-			zap.Stringer("nodeID", nodeID),
-			zap.Error(err),
-		)
+	err = n.issueTx(tx)
+	if err == nil {
```
Comment (StephenButtolph) [inline vms/platformvm/network/network.go:125]: I feel like this is a bit better just to make sure we never accidentally return the `err` here
```suggestion
	if err := n.issueTx(tx); err == nil {
```

Comment (dhrubabasu) [reply vms/platformvm/network/network.go]: went with your suggestion [e0f1b94](https://github.com/ava-labs/avalanchego/pull/2296/commits/e0f1b944a4410caff18dc728edb35800bb259ffe)

File: vms/platformvm/network/network.go [hunk 4]
```diff
@@ -109,60 +107,80 @@ func (n *network) AppGossip(ctx context.Context, nodeID ids.NodeID, msgBytes []b
...
 		return nil
 	}
-
-	// add to mempool
-	if err := n.IssueTx(ctx, tx); err != nil {
-		n.ctx.Log.Debug("tx failed verification",
-			zap.Stringer("nodeID", nodeID),
-			zap.Error(err),
-		)
+	if err = n.issueTx(tx); err == nil {
```
Comment (StephenButtolph) [inline vms/platformvm/network/network.go:124]:
```suggestion
	if err := n.issueTx(tx); err == nil {
```

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**